src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -196,6 +198,44 @@ public static String parseDayOfWeek(String shiftDay) throws ParseException {
         return trimmedStr;
     }
 
+    /**
+     * Parses a {@code String dayOfWeek} into an {@code DayOfWeek}.
+     * Leading and trailing whitespaces will be trimmed.
+     * This parser is not case sensitive.
+     *
+     * @throws ParseException if the given {@code dayOfWeek} is invalid.
+     */
+    public static String parseDayOfWeekAndTime(String shiftDay) throws ParseException {
+        String messageConstraints = "Valid input format: dayOfWeek-time:" + "List of valid dayOfWeek: "
+                + "Monday, Tuesday, Wednesday, Thursday, Friday, Saturday, Sunday. (Not case-sensitive)\n"
+                + "valid time formats: HH:mm in 24-hour format, such as 13:00";
+        requireNonNull(shiftDay);
+        String trimmedStr = shiftDay.trim().toLowerCase();
+        String[] strings = trimmedStr.split("-");
+        if (strings.length != 2) {
+            throw new ParseException(messageConstraints);
+        }
+        switch (strings[0]) {
+        case "monday":
+        case "tuesday":
+        case "wednesday":
+        case "thursday":
+        case "friday":
+        case "saturday":
+        case "sunday":
+            break;
+        default: throw new ParseException(messageConstraints);
+        }
```

Comment:
Hmm I think this is a weird use of a switch block. Maybe you could just use a normal check?
```suggestion
        List<String> strDays = Arrays.asList("Monday", "Tuesday", "Wednesday", "Thusday", "Friday", "Saturday", "Sunday" );
        if (!strDays.contains(strings[0])) {
            throw new ParseException(messageConstraints);
        }
```

Reply:
feel free to use isValidDayOfWeek() in Shift.java too maybe might be neater actl
```suggestion
        if (!isValidDayOfWeek(strings[0]) {
            throw new ParseException(messageConstraints);
        }
```

Reply:
> Hmm I think this is a weird use of a switch block. Maybe you could just use a normal check?

If you are using the Arrays method make sure to keep to lower case. I think it is better to use isValidDayOfWeek. Keeps it to single point of failure. Using the isValidDayOfWeek means we do not need to convert to lower case, though it is probably better to do so.